### ISSUE TITLE
Return type of missing classes is case insensitive (e.g. for an extension that isn't loaded)

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#240]
Observed: Whether or not an issue is emitted depends on the case of the return type. This includes Psalm's internal function signatures
Expected: Psalm's checks of class names should be case insensitive. (EDIT: added observed/expected)

```php
<?php

/** @return void */
function test_myclass(nonexistentclass $x) {
}

function make_myclass() : NonExistentClass {
    throw new RuntimeException("Solr not loaded");
}
function make_myclass2() : nonexistentclass {
    throw new RuntimeException("Solr not loaded");
}

test_myclass(make_myclass());  // This warns ERROR: InvalidArgument - src/case.php:16:14 - Argument 1 of test_myclass expects nonexistentclass, NonExistentClass provided 
test_myclass(make_myclass2());  // This doesn't
test_myclass(make_myclass());
test_myclass(make_myclass2());
```

**Comment by weirdan** [comment]
Theoretically, function and class names in PHP are case-insensitive, so either case could be accepted.

**Comment by muglug** [comment]
Depending on your autoloader and operating system, classes may well be case-sensitive, unfortunately.

**Comment by muglug** [comment]
Fixed by d455e24

**Comment by muglug** [comment]
Thanks!
